Bookde/Feature/MessageCenter/Sources/MessageCenter/UseCase/MessageUseCase.swift

```diff
@@ -14,13 +14,15 @@ public protocol MessageUseCase {
     func fetchCurrentUser() async -> AnyPublisher<UserChat, Never>
     func signOut() async -> AnyPublisher<Bool, Never>
     func fetchAllUser() async -> AnyPublisher<[UserChat], Never>
-    func send(toId:String, messgage: String)
+    func send(toId: String, message: String) async -> AnyPublisher<Bool, Never>
     func fetchMessage(toId: String) async -> AnyPublisher<[MessageModel], Never>
+    func fetchMessage(toId: String, completion: @escaping (Result<[MessageDTO],Error>) -> Void)
```

Comment:
I'm unsure of when to use @escaping closures versus async functions. @escaping closures work flawlessly, but async functions sometimes don't.